File: SDKs/amazonaws.com/apigatewayv2/2018-11-29/python/src/sdk/models/shared/updatedomainnameresponse.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class UpdateDomainNameResponseMutualTLSAuthentication:
    r"""UpdateDomainNameResponseMutualTLSAuthentication
    The mutual TLS authentication configuration for a custom domain name.
    """
    
    truststore_uri: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TruststoreUri') }})
    truststore_version: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TruststoreVersion') }})
    truststore_warnings: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TruststoreWarnings') }})
    

@dataclass_json
@dataclasses.dataclass
class UpdateDomainNameResponse:
    api_mapping_selection_expression: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ApiMappingSelectionExpression') }})
    domain_name: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DomainName') }})
    domain_name_configurations: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DomainNameConfigurations') }})
    mutual_tls_authentication: Optional[UpdateDomainNameResponseMutualTLSAuthentication] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MutualTlsAuthentication') }})
    tags: Optional[dict[str, str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    

File: SDKs/amazonaws.com/apigatewayv2/2018-11-29/typescript/src/sdk/models/shared/updatedomainnameresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// UpdateDomainNameResponseMutualTlsAuthentication
/** 
 * The mutual TLS authentication configuration for a custom domain name.
**/
export class UpdateDomainNameResponseMutualTlsAuthentication extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=TruststoreUri" })
  truststoreUri?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=TruststoreVersion" })
  truststoreVersion?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=TruststoreWarnings" })
  truststoreWarnings?: Record<string, any>;
}


export class UpdateDomainNameResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=ApiMappingSelectionExpression" })
  apiMappingSelectionExpression?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=DomainName" })
  domainName?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=DomainNameConfigurations" })
  domainNameConfigurations?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=MutualTlsAuthentication" })
  mutualTlsAuthentication?: UpdateDomainNameResponseMutualTlsAuthentication;

  @SpeakeasyMetadata({ data: "json, name=Tags" })
  tags?: Record<string, string>;
}
